src/data/InstanceDhisRepository.ts

```diff
@@ -201,10 +201,16 @@ export class InstanceDhisRepository implements InstanceRepository {
         return this.importAggregatedData("DELETE", dataPackage);
     }
 
-    public async importDataPackage(dataPackage: DataPackage): Promise<SynchronizationResult[]> {
+    public async importDataPackage(
+        dataPackage: DataPackage,
+        createAndUpdate: boolean
+    ): Promise<SynchronizationResult[]> {
         switch (dataPackage.type) {
             case "dataSets": {
-                const result = await this.importAggregatedData("CREATE_AND_UPDATE", dataPackage);
+                const result = await this.importAggregatedData(
+                    createAndUpdate ? "CREATE_AND_UPDATE" : "CREATE",
```

Comment:
Just as a note, when Duplicate detection is not enabled, duplicateStrategy is "ERROR" so the default importStrategy is now CREATE instead of CREATE_AND_UPDATE. That's possible a feature, not a bug, but I wanted to note it as it will change our behaviour until that point.